EC-84: Revise Configuration, Customers, eContent, Login and Logging (#5)
ezsystems/commerce-documentation: docs/guide/login_and_registration/forgot_change_password.md

# Forgot / change password

## Forgot password

If the customer does not remember the old password and cannot log in to the shop,
they can use the forgot password page.

![](../img/login_1.png)

### Workflow

> 1. The customer needs to provide the login or email address that should receive the link for password resetting.  

Reviewer: ```suggestion
1. The customer must provide the login or email address that should receive the link for password resetting.  
```

Additionally, if the shop expects the customer number as login, the customer can specify the customer number. The shop then searches through the users inside the Customer Center.
2. If the user exists, they receive an email with a unique URL. It includes the correct token.

> 3. When clicked, the user is redirected to a page where they can reset the password.

Reviewer: ```suggestion
3. When the user clicks the URL, they are redirected to a page where they can reset the password.
```

!!! caution

>     The user is able to change the password only if the token is provided and the user is anonymous (the browser has no logged-in session).

Reviewer: ```suggestion
    The user is able to change the password only if the token is provided and the user is anonymous (there is no logged-in session in the browser).
```

        - siso_core.data_processor.send_password_reminder_email
```

#### Service

> When building a form the shop checks if logging in with customer number is enabled.

Reviewer: ```suggestion
When building a form, the shop checks if logging in with a customer number is enabled.
```

}
```

#### Data Processors

> The `ses_forms.password_reminder_data_processor` data processor checks if the data provided in the form is correct and if the user exists.

Reviewer: ```suggestion
The `ses_forms.password_reminder_data_processor` data processor checks whether the data provided in the form is correct and if the user exists.
```

> The `siso_core.data_processor.send_password_reminder_email` data processor sends an email with link to change password functionality is sent if the User exists.

Reviewer: ```suggestion
If the User exists, the `siso_core.data_processor.send_password_reminder_email` data processor sends an email with a link to change the password.
```

> The link is be valid for a period of time specified in configuration:

Reviewer: ```suggestion
The link is valid for a time that you can specify in configuration:
```

``` 
siso_core.default.forget_password_token_valid: 1 hour
```

#### Use in templates

``` html+twig
<a href="{{ path('silversolutions_password_reminder', {'formTypeResolver': 'reminder'}) }}" class="link">{{ 'Forgot your password?'|st_translate }}</a>
```

The shop allows registrations of more than one account with the same email address.

> This is useful when eZ Commerce has to run a B2B and a B2C shop in one installation.

Reviewer: ```suggestion
This is useful when eZ Commerce runs a B2B and a B2C shop in one installation.
```

## Change password

If the customer is logged in but wants to change their password (for security reasons),
they can use the change password form located on the profile page.

![](../img/login_2.png)

#### Workflow

> 1. The shop checks if the user is logged in and no token is provided. Otherwise they cannot access the change password page.

Reviewer: ```suggestion
1. The shop checks whether the user is logged in and no token is provided. Otherwise they cannot access the change password page.
```

> 1. The user fills the form with new password and clicks send.

Reviewer: ```suggestion
1. The user provides a new password and submits the form.
```

> 1. The form is validated and new password is set for the customer.

Reviewer: ```suggestion
1. The form is validated and the password is updated for the customer.
```

}
```

> Then the form data is submitted and result is displayed.

Reviewer: ```suggestion
Then the form data is submitted and the result is displayed.
```
